clang/test/SemaSYCL/intel-fpga-loops.cpp

```diff
@@ -4,10 +4,10 @@
 void foo() {
   // expected-error@+1 {{intelfpga loop attributes must be applied to for, while, or do statements}}
   [[intel::ivdep]] int a[10];
-  // expected-error@+1 {{ loop attributes must be applied to for, while, or do statements}}
+  // expected-error@+1 {{intelfpga loop attributes must be applied to for, while, or do statements}}
```

Comment:
What causes this diagnostic to have the additional "intelfpga" in it?   I think we should change it to "Intel FPGA" that some other diagnostics below seem to be using.   (Perhaps as a different patch.)

Reply:
Sorry, I see why this is confusing! It's a drive-by NFC change.

The way `-verify` works is to see if it can find the text in the `{{}}` but the matching logic will happily ignore preceding or succeeding text. So that `expected-error` line would have also accepted a diagnostic like `fruit loop attributes must be applied to for, while, or do statements, as well as coco puffs`. According to the diagnostic itself, that diagnostic can start with `clang loop attributes`, so I added the `intelfpga` explicitly to ensure we were testing what we intended to test (hence why this is really NFC). FWIW, the usual community guidance is to use the full diagnostic text the first time in a test file, and shorten the text for subsequent diagnostics if the file is large and it'll save space. e.g., `intelfpga loop attributes must be applied to for, while, or do statements` for the first diagnostic, and `intelfpga loop attributes` for subsequent instances to save on disk space.

Allllll that said, I realized that the diagnostic is all kinds of bad for other reasons, so I'll be making changes in this area shortly (uncertain whether it'll impact the diagnostic text yet though). So I'm happy to either roll this change back or leave it in; your call.

Reply:
Thanks for the explanation!  Unless the new work undoes what you did here, I am fine with that being a separate patch.

Reply:
Excellent, thanks! It's unclear whether it will undo this particular change, so I'd say let's leave it in. If it does undo the change, it'll be touching a lot more than just that line in this test file anyway.